packages/core/src/com/communication.ts

```diff
@@ -98,7 +98,7 @@ export class Communication {
         this.registerMessageHandler(host);
         this.registerEnv(id, host);
         this.environments['*'] = { id, host };
-
+        this.messageIdPrefix = `c_${this.getEnvironmentId()}_${Math.random()}`;
```

Comment:
too long. the environment id is already on the message. also where did you encounter an issue with the callback ids?

Reply:
if the issue is for environment in the same runtime...you could add this to the constructor and use it as prefix. this will produce a small callback id.

```ts
Communication.instanceCount ||=0
Communication.instanceCount++;
```